cdc/scheduler/internal/v3/replication/replication_manager.go

```diff
@@ -512,6 +512,26 @@ func (r *Manager) AdvanceCheckpoint(
 	barrier *schedulepb.BarrierWithMinTs,
 	redoMetaManager redo.MetaManager,
 ) (newCheckpointTs, newResolvedTs model.Ts) {
+	limitBarrierWithRedo := func(newCheckpointTs, newResolvedTs *uint64) {
```

Comment:
Better use the return value, since it could be more clear, such as:
```go
limitWithRedo := func(newCheckpointTs, newResolvedTs uint64, needFlushMeta bool) (uint64, uint64) {
...
}

// calc newCheckpointTs and newResolvedTs

if redoMetaManager.Enabled() {
   return limitWithRedo(newCheckpointTs, newResolvedTs, needFlushMeta)
}
return newCheckpointTs, newResolvedTs
```